File: google_places_ios/ios/Classes/AutoCompleteExtensions.swift
import GooglePlaces

extension GMSAutocompletePrediction {
    func toJson() -> Dictionary<String, Any?>{
        return ["placeId": placeID,
                "primaryText": attributedFullText.string,
                "secondaryText": attributedSecondaryText?.string,
                "fullName": attributedFullText.string,
                "placeTypes": types]
    }
}

File: google_places_android/android/src/main/kotlin/com/google_places_android/AutoCompleteExtensions.kt
package com.google_places_android

import com.google.android.libraries.places.api.model.AutocompletePrediction

fun AutocompletePrediction.toJson(): Map<String, Any?> {
    return mapOf("placeId" to placeId,
            "primaryText" to getPrimaryText(null),
            "secondaryText" to getSecondaryText(null),
            "fullName" to getFullText(null),
            "placeTypes" to placeTypes.map { type -> type.value })
}